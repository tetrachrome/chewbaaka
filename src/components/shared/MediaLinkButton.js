/**
 * MediaLinkButton.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 07, 2020
 * Updated  : Jul 07, 2020
 */

import React from 'react';

import { Button, Icon } from 'semantic-ui-react'

function MediaLinkButton(props) {
  return (
    <div>
      <a href={props.href}>
        <Button icon color="blue">
          {props.title}
          <Icon name="file video" />
        </Button>
      </a>
    </div>
  );
}

export default MediaLinkButton;
