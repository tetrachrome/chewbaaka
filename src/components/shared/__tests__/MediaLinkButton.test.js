/**
 * MediaLinkButton.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 07, 2020
 * Updated  : Jul 07, 2020
 */

import React from 'react';

import { render, screen } from '@testing-library/react';

import MediaLinkButton from '../MediaLinkButton';

test('renders MediaLinkButton component', () => {
  const title = 'Watch "Walking with Chewbaaka" on YouTube';

  const href = "https://www.youtube.com/watch?v=WYjeEG06cjc";

  render(
    <MediaLinkButton href={href} title={title} />
  );

  const element = screen.getByText(title);
  expect(element).toBeInTheDocument();
});
