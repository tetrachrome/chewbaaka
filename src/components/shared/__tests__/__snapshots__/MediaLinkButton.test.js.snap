// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaLinkButton component snapshot 1`] = `
<div>
  <a
    href="https://www.youtube.com/watch?v=WYjeEG06cjc"
  >
    <button
      className="ui blue icon button"
      onClick={[Function]}
    >
      Watch "Walking with Chewbaaka" on YouTube
      <i
        aria-hidden="true"
        className="file video icon"
        onClick={[Function]}
      />
    </button>
  </a>
</div>
`;
